### PR TITLE
[quant] Explictly set default quantized engine instead of relying on the order of supported_qengines

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89804

Summary:
Fixes: https://github.com/pytorch/pytorch/issues/86404

Test Plan:
ossci + sandcastle
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D41635738](https://our.internmc.facebook.com/intern/diff/D41635738)

### CODE REVIEW

File: aten/src/ATen/Context.cpp [hunk 1]
```diff
@@ -295,8 +295,24 @@ bool Context::hasLAPACK() {
...
+    qengine = at::kONEDNN;
+#endif
+
+#ifdef USE_FBGEMM
+    if (fbgemm::fbgemmSupportedCPU()) {
+      qengine = at::kFBGEMM;
+    }
+#endif
+    return qengine;
+  }();
```
Comment (andrewor14) [inline aten/src/ATen/Context.cpp:314]: What's the purpose of using a lambda here? I feel it would be easier to read without it.

Comment (andrewor14) [reply aten/src/ATen/Context.cpp:314]: Also, is it possible to have return statements inside `ifdef` blocks? If so that would be cleaner I think

**Review by andrewor14 — APPROVED** [review]

File: aten/src/ATen/Context.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/Context.cpp:314]: this is to make sure the function is only run once during first call I think, sure I can move around return

Comment (jerryzh168) [reply aten/src/ATen/Context.cpp:314]: how to return in `ifdef`?

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41635738).

**Comment by jerryzh168** [comment]
> @pytorchbot revert -m "breaking tests https://hud.pytorch.org/pytorch/pytorch/commit/607ff6f4c10914a2a46bab90577cd083a6b3d46d https://github.com/pytorch/pytorch/actions/runs/3596841274/jobs/6058297637 trunk label didnt kick off workflows fast enough" -c nosignal

I remember checked the CI before actually, should this error block landing?

**Pull request closed without merging**